java/src/main/java/ai/rapids/cudf/DecimalUtils.java

```diff
@@ -142,7 +144,16 @@ public static ColumnVector greaterThan(ColumnView lhs, BigDecimal rhs) {
     // For example:
     //      100.2 > 100.19
     // If we rounded up the rhs 100.19 would become 100.2, and now 100.2 is not > 100.2
-    BigDecimal roundedRhs = rhs.setScale(-cvScale, BigDecimal.ROUND_DOWN);
+
+    // First we have to round the scalar (rhs) to the same scale as lhs.
+    // Because this is a greater-than, and it is rhs that we are rounding, we will round towards 0 (DOWN) in case rhs is
+    // positive and away from 0 (UP) if rhs is negative to make sure we always return the correct value.
```

Comment:
Can simplify to state we're rounding towards negative infinity, i.e.: always move to the next lower value when losing precision.